### ISSUE TITLE
[Kubernetes] If a "tls" section is specified in an Ingress, but a secret is not specified, Traefik will fail to load the Ingress

### ISSUE THREAD

**Issue by DrEsteban** [containous/traefik#4280]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

We deployed Traefik with a "default certificate" specified, so that we didn't have to specify a TLS cert on every Ingress resource.  (We have a wildcard TLS cert so we'd prefer to just specify in one place - at the Traefik level.)  Our Ingress `spec` sections look like the following:

```yaml
spec:
  rules:
  - host: foo.bar.com
    http:
      paths:
        <paths...>
  tls:
  - hosts:
    - foo.bar.com
```
^ Notice the "tls" section, which simply has the host listed but not any secret info. This is because we are relying on the default certificate.

### What did you expect to see?

Traefik would load the Ingress rule, and would present its default certificate since a different one wasn't specified in the Ingress.  For reference, this works as expected with the Nginx Ingress controller.

### What did you see instead?

Traefik error logs:
```
{"level":"error","msg":"Error configuring TLS for ingress mynamespace/myservice: secret mynamespace/ does not exist","time":"2018-12-07T19:38:38Z"}
```
^Notice the empty secret name.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.5
Codename:     maroilles
Go version:   go1.11.2
Built:        2018-12-03_11:01:00AM
OS/Arch:      linux/amd64
```
(Note, also repros on v1.7.4)

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
# traefik.toml
logLevel = "INFO"
defaultEntryPoints = ["http","https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
  compress = true
    [entryPoints.http.redirect]
      regex = "^http://(.*)"
      replacement = "https://$1"
      permanent = true
  [entryPoints.https]
  address = ":443"
  compress = true
    [entryPoints.https.tls]
      minVersion = "VersionTLS12"
      cipherSuites = [
        "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
        "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
        "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
        "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
        "TLS_RSA_WITH_AES_128_GCM_SHA256",
        "TLS_RSA_WITH_AES_256_GCM_SHA384"
      ]
    [entryPoints.https.tls.defaultCertificate]
      CertFile = "/ssl/tls.crt"
      KeyFile = "/ssl/tls.key"
  [entryPoints.traefik]
  address = ":8080"
[kubernetes]
[traefikLog]
  format = "json"
[api]
  entryPoint = "traefik"
  dashboard = true
```

**Comment by coolboy190** [comment]
We've noticed the same issue and I agree it should be fixed. We were earlier using an nginx ingress controller that used to work with the tls section but no secretName.

**Comment by dduportal** [comment]
Hi @DrEsteban and @coolboy190 , thanks for your feedbacks.

The documentation helps you on configuring the TLS certificates: https://docs.traefik.io/user-guide/kubernetes/#add-a-tls-certificate-to-the-ingress .
It is mandatory to use a Kubernetes Secret for providing the certificates.

There is one exception: if you specify the default certificate, as pointed by @dtomcej in https://github.com/containous/traefik/issues/4276#issuecomment-445077968 , and do NOT specify the TLS section in Kubernetes, then this default will be used (which sounds like the use case). The `tls` section being aimed to handle per-host certificate (or dynamic ones).

We understand that other Ingress controllers allows this, to follow the Kubernetes's spec. But it makes the certificates in "clear" on the container filesystem (not mentioning the constraint to reschedule this is creating). We have scheduled this improvement as a "P3" element.

**Comment by dtomcej** [comment]
Closed by #4340
